apypie/api.py

```diff
@@ -60,7 +62,35 @@ def __init__(self, **kwargs):
         if kwargs.get('username') and kwargs.get('password'):
             self._session.auth = (kwargs['username'], kwargs['password'])
 
-        self.apidoc = self._load_apidoc()
+        self._apidoc = self._load_apidoc()
+
+    @property
+    def apidoc(self):
+        if self._apidoc is None:
+            self._apidoc = self._load_apidoc()
+        return self._apidoc
+
+    @property
+    def apidoc_cache_file(self):
+        return os.path.join(self.apidoc_cache_dir, '{0}{1}'.format(self.apidoc_cache_name, self.cache_extension))
+
+    def _set_default_name(self, default='default'):
+        """find the newest file in the cachedir if any"""
+        cache_file = sorted(glob.iglob(os.path.join(self.apidoc_cache_dir, '*{}'.format(self.cache_extension))), key=os.path.getctime, reverse=True)
+        if cache_file:
+            return os.path.basename(cache_file[0]).replace(self.cache_extension, '')
+        else:
+            return default
+
+    def update_cache(self, cache_name):
+        if cache_name is not None and cache_name != self.apidoc_cache_name:
+            self.clean_cache()
+            self.apidoc_cache_name = cache_name
```

Comment:
Do we want to assert `cache_name` doesn't include a `/` so it can't be in a subdirectory or traverse somewhere?